tests/playwright-test/codegen.spec.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { test, expect } from './playwright-test-fixtures';
+import { generatePlaywrightRequestCall } from '../../packages/playwright-core/src/utils/isomorphic/codegen';
+
+test('generatePlaywrightRequestCall', () => {
+  expect(generatePlaywrightRequestCall({
+    url: 'http://example.com/foo?bar=baz',
+    method: 'GET',
+    headers: [{ name: 'User-Agent', value: 'Mozilla/5.0' }, { name: 'Date', value: '2021-01-01' }],
+    httpVersion: '1.1',
+    cookies: [],
+    queryString: [],
+    headersSize: 0,
+    bodySize: 0,
+    comment: '',
+  }, 'foo')).toEqual(`
+await page.request.get('http://example.com/foo', {
```

Comment:
What about the other languages? Also, most of your code is around formatting, please add more tests to cover edge condition. This test belongs to `tests/library/unit` as it does not use e2e capabilities too.

Reply:
> What about the other languages?

I wanted to keep things simple so I only started with Javascript. We can extend it when users ask for it.